libglusterfs/src/dict.c

```diff
@@ -3521,9 +3516,11 @@ dict_unserialize_specific_keys(char *orig_buf, int32_t size, dict_t **fill,
     int32_t vallen = 0;
     int32_t hostord = 0;
     xlator_t *this = NULL;
+    int *keylenarr = NULL;
```

Comment:
We could create an array in stack instead of using dynamic memory:

```suggestion
    int32_t keylenarr[totkeycount];
```